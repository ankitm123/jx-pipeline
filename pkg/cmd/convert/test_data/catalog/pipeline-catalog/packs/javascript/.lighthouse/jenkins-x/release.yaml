apiVersion: tekton.dev/v1
kind: PipelineRun
metadata:
  annotations:
    lighthouse.jenkins-x.io/prependStepsURL: https://raw.githubusercontent.com/jenkins-x/jx3-pipeline-catalog/60bed6408732c1eda91a15713f51a9f97dcb1757/tasks/git-clone/git-clone.yaml
  name: release
spec:
  pipelineSpec:
    tasks:
    - name: from-build-pack
      computeResources: {}
      taskSpec:
        metadata: {}
        stepTemplate:
          env:
          - name: NPM_CONFIG_USERCONFIG
            value: /tekton/home/npm/.npmrc
          computeResources:
            requests:
              cpu: 400m
              memory: 512Mi
          volumeMounts:
          - name: npmrc
            mountPath: /tekton/home/npm
          workingDir: /workspace/source
        steps:
        - image: gcr.io/jenkinsxio/jx-release-version:1.0.42
          name: next-version
          computeResources: {}
          script: |
            #!/usr/bin/env bash
            jx-release-version > VERSION
        - image: gcr.io/jenkinsxio/jx-boot:3.1.156
          name: jx-variables
          computeResources: {}
          script: |
            #!/usr/bin/env sh
            jx gitops variables
        - image: node:12-slim
          name: build-npm-install
          computeResources: {}
          script: |
            #!/bin/sh
            npm install
        - image: node:12-slim
          name: build-npm-test
          computeResources: {}
          script: |
            #!/bin/sh
            CI=true DISPLAY=:99 npm test
        - image: gcr.io/kaniko-project/executor:debug-v1.3.0
          name: build-container-build
          computeResources: {}
          script: |
            #!/busybox/sh
            source .jx/variables.sh
            cp /tekton/creds-secrets/tekton-container-registry-auth/.dockerconfigjson /kaniko/.docker/config.json
            /kaniko/executor $KANIKO_FLAGS --context=/workspace/source --dockerfile=Dockerfile --destination=$DOCKER_REGISTRY/$DOCKER_REGISTRY_ORG/$APP_NAME:$VERSION
        - image: gcr.io/jenkinsxio/jx-changelog:0.0.25
          name: promote-changelog
          computeResources: {}
          script: |
            #!/usr/bin/env sh
            source .jx/variables.sh

            if [ -d "charts/$REPO_NAME" ]; then
            sed -i -e "s/^version:.*/version: $VERSION/" ./charts/$REPO_NAME/Chart.yaml
            sed -i -e "s/repository:.*/repository: $DOCKER_REGISTRY\/$DOCKER_REGISTRY_ORG\/$APP_NAME/" ./charts/$REPO_NAME/values.yaml
            sed -i -e "s/tag:.*/tag: $VERSION/" ./charts/$REPO_NAME/values.yaml;
            else echo no charts; fi

            git commit -a -m "chore: release $VERSION" --allow-empty
            git tag -fa v$VERSION -m "Release version $VERSION"
            git push origin v$VERSION

            jx changelog create --version v${VERSION}
        - image: gcr.io/jenkinsxio/jx-boot:3.1.156
          name: promote-helm-release
          computeResources: {}
          script: |
            #!/usr/bin/env sh
            jx gitops helm release
        - image: gcr.io/jenkinsxio/jx-promote:0.0.179
          name: promote-jx-promote
          computeResources: {}
          script: |
            #!/usr/bin/env sh
            source .jx/variables.sh
            jx promote -b --all --timeout 1h --no-poll
        volumes:
        - name: npmrc
          secret:
            secretName: npmrc
            optional: true
  taskRunTemplate:
    serviceAccountName: tekton-bot
  timeouts:
    pipeline: 240h0m0s
