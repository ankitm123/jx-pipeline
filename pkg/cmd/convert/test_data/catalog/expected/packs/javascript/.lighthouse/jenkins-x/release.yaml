apiVersion: tekton.dev/v1
kind: PipelineRun
metadata:
  creationTimestamp: null
  name: release
spec:
  pipelineSpec:
    tasks:
    - name: from-build-pack
      taskSpec:
        metadata: {}
        spec: null
        stepTemplate:
          computeResources:
            requests:
              cpu: 400m
              memory: 512Mi
          env:
          - name: NPM_CONFIG_USERCONFIG
            value: /tekton/home/npm/.npmrc
          image: uses:jenkins-x/jx3-pipeline-catalog/tasks/javascript/release.yaml@versionStream
          volumeMounts:
          - mountPath: /tekton/home/npm
            name: npmrc
          workingDir: /workspace/source
        steps:
        - computeResources: {}
          image: uses:jenkins-x/jx3-pipeline-catalog/tasks/git-clone/git-clone.yaml@versionStream
          name: ""
        - computeResources: {}
          name: next-version
        - computeResources: {}
          name: jx-variables
        - computeResources: {}
          name: build-npm-install
        - computeResources: {}
          name: build-npm-test
        - computeResources: {}
          name: build-container-build
        - computeResources: {}
          name: promote-changelog
        - computeResources: {}
          name: promote-helm-release
        - computeResources: {}
          name: promote-jx-promote
        volumes:
        - name: npmrc
          secret:
            optional: true
            secretName: npmrc
  taskRunTemplate:
    serviceAccountName: tekton-bot
  timeouts:
    pipeline: 240h0m0s
status: {}
